src/diffusers/models/autoencoders/autoencoder_kl_cogvideox.py

```diff
@@ -993,8 +1037,34 @@ def encode(
             return (posterior,)
         return AutoencoderKLOutput(latent_dist=posterior)
 
+    def _decode(self, z: torch.Tensor, return_dict: bool = True) -> Union[DecoderOutput, torch.Tensor]:
+        if self.use_tiling and (z.shape[-1] > self.tile_latent_min_width or z.shape[-2] > self.tile_latent_min_height):
+            return self.tiled_decode(z, return_dict=return_dict)
+
+        num_latent_frames = z.shape[2]
+        dec = []
+        for i in range(num_latent_frames // 2):
```

Comment:
Why the `//2`?

Reply:
We are trying to decode 2 frames at a time (start_frame:end_frame), so we divide total latent frames by 2. We can decode more frames at a time, but will lead to memory increase and may not play well with the context parallel cache implementation but that's something I need to try. Could probably make it a configurable parameter in enable_tiling if it plays well with CP cache?

Reply:
Yeah let's do that. If it doesn't play well, let's add a comment.